### PR TITLE
Allow skipping 2D class selection

### PR DESCRIPTION
**Pull request by d-j-hatton**
Skip if parameter `class2d_fraction_of_classes_to_remove` is 0

### CODE REVIEW

File: src/relion/pipeline/__init__.py [hunk 1]
```diff
@@ -697,7 +699,10 @@ def classification(
                         class_scores[class_scores >= 0], fraction_of_classes_to_remove
                     )
 
-                if not self.job_paths_batch.get(class2d_type):
+                if (
+                    not self.job_paths_batch.get(class2d_type)
+                    and fraction_of_classes_to_remove
+                ):
```
Comment (stephen-riggs) [inline src/relion/pipeline/__init__.py:705]: I think this eliminates the first round of 2D classification entirely. Without class selection we only need to skip lines 721 to 732, and the current line 708 probably should move to there as well. (i.e. only an if on occurrences of "relion.select.class2D")

File: src/relion/pipeline/__init__.py [hunk 2]
```diff
@@ -862,22 +867,27 @@ def classification(
...
-                                    "angpix": angpix,
-                                    "boxsize": boxsize,
-                                },
-                            )
-                            class3d_thread.start()
-                        self._queues["class3D"][iteration].put(split_file)
-                        # mark this split as the last one completed and run
-                        last_completed_split += 1
+                if (
+                    len(split_file_column) == self.options.batch_size
```
Comment (stephen-riggs) [inline src/relion/pipeline/__init__.py:871]: Will this throw an error that split_file_column is not defined?

**Review by stephen-riggs — CHANGES_REQUESTED** [review]
We'll also need if statements to remove class selection for lines 762 to 791 and 806 to 823 when the threshold is set to zero

**Review by stephen-riggs — APPROVED** [review]
Changes look good.
Only other thing might be the if statement starting on line 677. I'm not certain how it behaves so is it worth adding "and fraction_of_classes_to_remove" to it?

**Comment by d-j-hatton** [comment]
Thanks for spotting those

**Pull request merged**